### PR TITLE
fix(FilterManager): fixes bug with the period filter from-To and other

### PR DESCRIPTION
**Pull request by ferdyrod**
**JIRA ISSUE**: [ANDROAPP-2512](https://jira.dhis2.org/browse/ANDROAPP-2512)

### CODE REVIEW

File: app/src/main/java/org/dhis2/utils/filters/PeriodFilterHolder.java [hunk 1]
```diff
@@ -84,18 +84,8 @@ private void setListeners(FilterPeriodBinding periodLayout) {
...
-            }
-        });
-        periodLayout.other.setOnClickListener(view -> {
-            if (periodLayout.other.isChecked()) {
-                updateSelection(R.id.other);
-                FilterManager.getInstance().addPeriodRequest(FilterManager.PeriodRequest.OTHER);
-            }
-        });
+        periodLayout.fromTo.setOnCheckedChangeListener(this);
+        periodLayout.other.setOnCheckedChangeListener(this);
```
Comment (Balcan) [inline app/src/main/java/org/dhis2/utils/filters/PeriodFilterHolder.java]: This was using the onclicklistener to be able to select again these filters even if they are already selected.
Have you test to click them once they are checked?

Comment (ferdyrod) [reply app/src/main/java/org/dhis2/utils/filters/PeriodFilterHolder.java]: Ok, will have to change it. Didn't know that that was a possibility.

Comment (Balcan) [reply app/src/main/java/org/dhis2/utils/filters/PeriodFilterHolder.java]: Yep, it was discussed in this [issue](https://jira.dhis2.org/browse/ANDROAPP-2401)

**Comment by Balcan** [comment]
Tested!

**Pull request merged**